src/test/scala/CoreServerTests.scala

```diff
@@ -0,0 +1,340 @@
+        })
+        messages_future2.onComplete({
+          case Success(verRes) => succeed
+          case Failure(_) => fail()
+        })
+        messages_future3.onComplete({
+          case Success(verRes) => succeed
+          case Failure(_) => fail()
+        })
+      }
+
+      "be able to execute 'stop()' without exceptions" in {
+        core.stop()
+      }
+    }
+    "verifying multiple files with caching enabled" should {
+      val file1 = sumFile
+      val file2 = verifiableFile
+      val astGen1 = new AstGenerator(file1, silent_logger)
+      val astGen2 = new AstGenerator(file2, silent_logger)
+
+      val config = new ViperConfig(List())
+      config.verify()
+      val core = new ViperCoreServer(config)
+      core.start()
+
+      var handler1: VerificationJobHandler = null
+      var handler2: VerificationJobHandler = null
+      var handler3: VerificationJobHandler = null
+      "be able to execute 'verify()' repeatedly without exceptions" in {
+        handler1 = core.verify(file1, SiliconConfig(List()), astGen1.viper_ast.get)
+        handler2 = core.verify(file2, SiliconConfig(List()), astGen2.viper_ast.get)
+        handler3 = core.verify(file2, SiliconConfig(List()), astGen2.viper_ast.get)
+        assert(handler1 != null)
+        assert(handler2 != null)
+        assert(handler3 != null)
+      }
+
+      "have 'verifiy()' return JobHandlers with unique positive ids when executed " in {
+        assert(handler1.id == 0)
+        assert(handler2.id == 1)
+        assert(handler3.id == 2)
+      }
+
+      var messages_future1: Future[Seq[Message]] = null
+      var messages_future2: Future[Seq[Message]] = null
+      var messages_future3: Future[Seq[Message]] = null
+      "be able to have 'getMessagesFuture()' return a future of a sequence of Viper messages." in {
+        messages_future1 = core.getMessagesFuture(handler1.id)
+        messages_future2 = core.getMessagesFuture(handler2.id)
+        messages_future3 = core.getMessagesFuture(handler3.id)
+        assert(messages_future1 != null)
+        assert(messages_future2 != null)
+        assert(messages_future3 != null)
+      }
+
+      "eventually see the future returned from 'getMessagesFuture()' completed sucessfully" in {
+        while (!messages_future1.isCompleted && !messages_future2.isCompleted && !messages_future3.isCompleted) {
+          Thread.sleep(500)
+        }
+        messages_future1.onComplete({
+          case Success(verRes) => succeed
+          case Failure(_) => fail()
+        })
+        messages_future2.onComplete({
+          case Success(verRes) => succeed
+          case Failure(_) => fail()
+        })
+        messages_future3.onComplete({
+          case Success(verRes) => succeed
+          case Failure(_) => fail()
+        })
+      }
+
+      "be able to execute 'stop()' without exceptions" in {
+        core.stop()
+      }
+    }
+  }
+  "An instance of ViperCoreServer" when {
```

Comment:
Couln't we use multiple ```"bla-bla" should { ... }``` entries in a single ```"An instance of ViperCoreServer" when``` scope?

Reply:
Sure. In some places this is already the case. I didn't put everything into one "when" block to keep the test cases more organized.